api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -61,6 +61,13 @@ default void deleteFile(OutputFile file) {
     deleteFile(file.location());
   }
 
+  /**
+   * Delete a folder at a given path.
+   */
+  default void deleteFolder(String path) {
+    deleteFile(path);
+  }
+
```

Comment:
Do we need this?

Reply:
Currently only empty folders could be dropped as the 'isRecursive' flag is hardcoded to false in HadoopFileIO.
If we want to keep FileIO intact, we can then change new logic in HiveCatalog.dropTable() to check if the FileIO is of HadoopFileIO and use its interface to invoke dropFolder() (or could be named dropRecursive()) instead of through FileIO.

Reply:
Maybe `SupportsPrefixOperations.deletePrefix` could help us here.
See the `HadoopFileIO.deletePrefix` implementation.

Reply:
Aaah, this code is relatively new, I didn't have it in my local copy. HadoopFileIO.deletePrefix() is actually the same I implemented now, so apparently someone also needed this but was a bit faster than me :)